src/test/java/spoon/test/architecture/SpoonArchitectureEnforcerTest.java

```diff
@@ -212,6 +216,34 @@ public boolean matches(CtClass element) {
 	}
 
 	@Test
+	public void testInterfacesAreCtScannable() {
+		// contract: all non-leaf interfaces of the metamodel should be visited by CtInheritanceScanner
```

Comment:
This test checks that there is a scan method. It is good test. But there is not checked that visitXxx method calls all scan methods for each interface implemented by that non abstract class. I suggest to add this test too. May be as part of another PR?
WDYT?

Reply:
I thought it was the goal of https://github.com/INRIA/spoon/blob/65c1b2974af8103b3b44f6f6306a310e6552e868/src/test/java/spoon/reflect/visitor/CtInheritanceScannerTest.java#L118